lib/src/server/server.dart

```diff
@@ -279,10 +284,19 @@ class Server extends ConnectionServer {
         settings: http2ServerSettings,
       );
 
+      InternetAddress? remoteAddress;
```

Comment:
Maybe encapsulate this into a helper:

```dart
extension on Socket {
  /// Tries to get a remote address if the socket is still open otherwise returns `null`.
  InternetAddress? remoteAddressOrNull {
    // ...
  }
}
```